### PR TITLE
Fixes #168 - Dark theme support

### LINKED ISSUE #168: Theme: Dark Mode Support
Is it possible to have Dark Mode Theme for the Report? Currently there is a provision to override the styles through 'overrideStyle' option. but this doesn't give flexibility to switch themes from dark to light theme. Can we also give provision to add a custom script which is inserted during html generation and we can combine both 'overrideStyle' option along with custom script to toggle between the themes?

**Custom Script: (to be embedded in index.html)**
const toggle = document.getElementById("toggle");
const refresh = document.getElementById("refresh");
const theme = window.localStorage.getItem("theme");

/* checks if the theme stored in localStorage is dark
if yes apply the dark theme to the body */
if (theme === "dark") document.body.classList.add("dark");

// event listener stops when the change theme button is clicked
toggle.addEventListener("click", () => {
   document.body.classList.toggle("dark");
   if (theme === "dark") {
     window.localStorage.setItem("theme", "light");
   } else window.localStorage.setItem("theme", "dark");
});

refresh.addEventListener("click", () => {
   window.location.reload();
});

**Custom Style.css:**
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}
/* Definition of colors */
body {
  /* text colors */
  --text-white: #ffffff;
  --text-dark: #142136;

  /* background colors */
  --bg-grey-light: #f5f5f5;
  --bg-white: #ffffff;
  --bg-blue-dark: #142136;
  --bg-indigo: #6366f1;

  font-family: "Inter", sans-serif;
  line-height: 1.7;
  background-color: var(--bg-grey-light);
}
.dark {
  --text-white: #e6e6e6;
  --text-dark: #ffffff;
  --bg-grey-light: #142136;
  --bg-white: #22395d;
  --bg-blue-dark: #142136;
  --bg-indigo: #7577e1;
}
.container {
  max-width: 600px;
  margin: 40px auto;
  display: flex;
  padding: 20px;
  flex-direction: column;
}
.text-wrapper {
  width: 100%;
  padding: 20px;
  background-color: var(--bg-white);
  margin-bottom: 40px;
  border-radius: 10px;
}
.paragraph {
  font-size: 16px;
  color: var(--text-dark);
}
.heading {
  font-size: 40px;
  letter-spacing: 1px;
  font-weight: 900;
  margin-bottom: 40px;
  color: var(--text-dark);
}
.buttons {
  width: 100%;
  display: flex;
  justify-content: space-between;
  margin-bottom: 40px;
}
.button {
  width: 200px;
  padding: 5px;
  height: 40px;
  border: none;
  border-radius: 10px;
  font-family: inherit;
  cursor: pointer;
  background-color: var(--bg-indigo);
  color: var(--text-white);
  font-size: 16px;
  font-weight: 400;
  text-transform: capitalize;
}

### PR DESCRIPTION
**Pull request by Baskercarrer**
Fixes for #168 

* BG Colors are applied thought element level. 

_Note: I know it is not a professional way. Please feel free to decline it if not suits._

image

### CODE REVIEW

**Comment by Baskercarrer** [comment]
thanks for the suggestion

**Comment by BaCaRoZzo** [comment]
@Baskercarrer not sure if it works on the title but it surely works in [the body of the message](https://github.com/wswebcreation/multiple-cucumber-html-reporter/pull/182#issue-1245462719). So you should add it there. :+1:

**Review by wswebcreation — CHANGES_REQUESTED** [review]
Wow @Baskercarrer 

This looks really awesome, thanks for all the effort, some small requests

Can you adjust the selectbox to this
image
with this code for darkmode
```css
div.dataTables_wrapper div.dataTables_length select {
    width: 75px;
    display: inline-block;
    background: #3b3b3b;
    border-color: #858585;
    color: #eee;
}
```

Same for the search box
image
```css
div.dataTables_wrapper div.dataTables_filter input {
    margin-left: 0.5em;
    display: inline-block;
    width: auto;
    background: #3b3b3b;
    border-color: #858585;
    color: #eee;
}
```

Can you also change the table a bit more into darkish mode?
Something like this

image

where the table header background is `#`3b3b3b` and all blue colors are `#eee`?

The we only have the pager left, here's an example
image

But maybe you have a better idea about this

Then when I look at the feature pages I see this
image

I'm not sure what the issue is here, but it looks like the body doesn't have the right darkmode. Can you also check the colors there?

Again, thanks for all your hard work on this

**Comment by Baskercarrer** [comment]
@wswebcreation - made a commit as per your suggestion. Could you please check one time?
Note: Table header color is chosen by myself and am happy to get color code if you are have something else to add. 
* Now colors are applied by CSS not by JS. Hope it will be faster than before i believe.

File: templates/style.css [hunk 1]
```diff
@@ -497,7 +495,16 @@ body.darkmode tabl {
...
     border-color: gray;
 }
+
+body.darkmode .form-control {
+    background-color: #3e3939 !important;
+    color: #c3c3c3;
+}
+body.darkmode li[id*=features]>a {
+    background-color: #3e3939 !important;
+
```
Comment (wswebcreation) [inline templates/style.css:509]: Maybe remove this extra line break
```suggestion
```

**Review by wswebcreation — CHANGES_REQUESTED** [review]
Thanks man, just a few small last comments

Can you remove the hover effect for darkmode on the chevrons?
image
```css
.darkmode .panel_toolbox > li > a:hover {
   background: transparent;
}
```

and change this button, on the feature page, to a color that is more in line with the rest?
image


something like this
image
```css
.darkmode .btn-info {
    color: #eee; 
    background-color: #337ab7;
    border-color: #337ab7;
}
```

If you agree and can push them then I can also merge it and release a new version

**Comment by Baskercarrer** [comment]
its my bad to not noticing such things. Anyway am done.

**Review by wswebcreation — APPROVED** [review]
LGTM now, thanks for all your work here!

**Pull request merged**